src/sentry/incidents/logic.py

```diff
@@ -751,6 +761,15 @@ def update_alert_rule(
         if deleted_subs:
             bulk_delete_snuba_subscriptions(deleted_subs)
 
+    if user:
+        create_audit_entry_from_user(
+            user,
+            organization_id=alert_rule.organization_id,
+            target_object=alert_rule.id,
+            data=alert_rule.get_audit_log_data(),
+            event=AuditLogEntryEvent.ALERT_RULE_EDIT,
+        )
```

Comment:
I'd probably put this near line 681 where we create the alert rule activity. I have a feeling something happens after that point that is causing your failing test.